addOns/encoder/src/main/javahelp/org/zaproxy/addon/encoder/resources/help/contents/encoder.html

```diff
@@ -206,6 +206,29 @@ <H4>Reverse</H4>
 <H4>To Upper Case</H4>
 Converts the input to all upper case characters.
 
+<H3>Miscellaneous</H3>
+
+<H4>PowerShell Encode</H4>
+Converted to UTF-16LE and base64 encoded. Equivalent to one of the following examples (encoding "dir"):
+<p>
+<code> printf "dir"| iconv -t UTF-16LE | base64 -w 0 -</code>
+<br>
+<pre>
+<code>
+$command = 'dir'
+$bytes = [System.Text.Encoding]::Unicode.GetBytes($command)
+$encodedCommand = [Convert]::ToBase64String($bytes)
+echo $encodedCommand
+</code>
+</pre>
+<br>
+Resulting in: <code>ZABpAHIA</code>.
+<p>
+Which can be leveraged via something like the following (where <code>$encodedCommand</code> is set to the encoded value <code>ZABpAHIA</code>):<br>
+<code>powershell -exec bypass -enc $encodedCommand</code>
+<p>
+Note: The standard Base64 Decoder seems to handle values encoded by this method just fine.
```

Comment:
This is not true (for all characters) since the charset used is different.

Reply:
Okay, I can remove the note. I didn't look for any edge cases but in my simple testing it did work.

Reply:
It's not an edge case, any characters outside ASCII (? maybe others) would be encoded differently, try with e.g.: `あ`.

Reply:
"edge" was a poor choice of words. I did only try ascii examples.

Shall I add a decoder as well?

Other than this is there other feedback/review :wink:

Reply:
I don't have more comments.

IMHO we don't need a new decoder (nor encoder for that matter), this is just Base64 with custom charset (which we allow to choose. Yes, UTF-16* are not yet there). We already talked about allowing the Base64 De/Encoder to be more flexible by allowing to provide the charset rather than use the singleton form the options but those are other long term changes. Once those are in place this encoder can just be added as a specific Base64 one.

(Note that I'm not saying we shouldn't add a custom name, it's good to have that but the implementations themselves should be using the Base64 de/encoder.)

Reply:
Okay so for the time being I'll just remove the help note?

Reply:
Yes, I don't think that note is accurate.

Reply:
Actually, maybe be more explicit and call it `powershellencoder`? (If we decide to add a decoder later.)

Reply:
That makes sense. Will do.